File: src/main/java/com/yupi/template/model/entity/PromptTemplate.java
package com.yupi.template.model.entity;

import com.mybatisflex.annotation.Column;
import com.mybatisflex.annotation.Id;
import com.mybatisflex.annotation.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.io.Serial;
import java.io.Serializable;
import java.time.LocalDateTime;

/**
 * 提示词模板实体
 *
 * @author <a href="https://codefather.cn">编程导航学习圈</a>
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Table("prompt_template")
public class PromptTemplate implements Serializable {

    @Serial
    private static final long serialVersionUID = 1L;

    /**
     * 模板唯一标识
     */
    @Id
    private String id;

    /**
     * 用户ID(预设模板为NULL)
     */
    @Column("userId")
    private Long userId;

    /**
     * 模板名称
     */
    @Column("name")
    private String name;

    /**
     * 模板描述
     */
    @Column("description")
    private String description;

    /**
     * 策略类型: direct/cot/role_play/few_shot
     */
    @Column("strategy")
    private String strategy;

    /**
     * 模板内容(支持占位符)
     */
    @Column("content")
    private String content;

    /**
     * 变量列表(JSON数组)
     */
    @Column("variables")
    private String variables;

    /**
     * 分类
     */
    @Column("category")
    private String category;

    /**
     * 是否为预设模板(1-预设 0-自定义)
     */
    @Column("isPreset")
    private Integer isPreset;

    /**
     * 使用次数
     */
    @Column("usageCount")
    private Integer usageCount;

    /**
     * 是否启用(1-启用 0-禁用)
     */
    @Column("isActive")
    private Integer isActive;

    /**
     * 创建时间
     */
    @Column("createTime")
    private LocalDateTime createTime;

    /**
     * 更新时间
     */
    @Column("updateTime")
    private LocalDateTime updateTime;

    /**
     * 是否删除
     */
    @Column(value = "isDelete", isLogicDelete = true)
    private Integer isDelete;
}

File: python-backend/app/models/prompt_template.py
"""
提示词模板数据库模型
@author <a href="https://codefather.cn">编程导航学习圈</a>
"""
from datetime import datetime
from sqlalchemy import Column, String, BigInteger, Text, Integer, DateTime
from sqlalchemy.orm import declarative_base

Base = declarative_base()


class PromptTemplate(Base):
    """
    提示词模板表模型
    """
    __tablename__ = "prompt_template"

    id = Column(String(36), primary_key=True, comment="模板唯一标识")
    user_id = Column("userId", BigInteger, nullable=True, comment="用户ID(预设模板为NULL)")
    name = Column("name", String(100), nullable=False, comment="模板名称")
    description = Column("description", Text, nullable=True, comment="模板描述")
    strategy = Column("strategy", String(50), nullable=False, comment="策略类型: direct/cot/role_play/few_shot")
    content = Column("content", Text, nullable=False, comment="模板内容(支持占位符)")
    variables = Column("variables", String(500), nullable=True, comment="变量列表(JSON数组)")
    category = Column("category", String(50), nullable=True, comment="分类")
    is_preset = Column("isPreset", Integer, nullable=False, default=0, comment="是否为预设模板(1-预设 0-自定义)")
    usage_count = Column("usageCount", Integer, nullable=False, default=0, comment="使用次数")
    is_active = Column("isActive", Integer, nullable=False, default=1, comment="是否启用(1-启用 0-禁用)")
    create_time = Column("createTime", DateTime, nullable=False, default=datetime.now, comment="创建时间")
    update_time = Column("updateTime", DateTime, nullable=False, default=datetime.now, onupdate=datetime.now, comment="更新时间")
    is_delete = Column("isDelete", Integer, nullable=False, default=0, comment="是否删除")

    def __repr__(self):
        return f"<PromptTemplate(id={self.id}, name={self.name})>"
